### PR TITLE
deps: update axe-core to 3.4.1

### PR DESCRIPTION
**Pull request by jayaddison**
**Summary**
Updates `axe-core==3.4.1` following its release on 2019-12-18 ([release tag](https://github.com/dequelabs/axe-core/releases/tag/v3.4.1)).

### CODE REVIEW

**Comment by jayaddison** [comment]
Upgrading the dependency here pushes the `lighthouse-dt-bundle.js` file size over the configured limit.  I'm not sure what the policy on increasing that is, so I'll leave the PR as-is.

Error snippets from the build log:
```
 FAIL  ./dist/lighthouse-dt-bundle.js: 401.52KB > maxSize 400KB (gzip)
...
The command "yarn bundlesize" exited with 1.
```

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js [hunk 1]
```diff
@@ -312,23 +312,6 @@ const expectations = [
             ],
           },
         },
-        'form-field-multiple-labels': {
```
Comment (jayaddison) [inline lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js]: It's hinted at in one of the commit messages, but to call out why this test case was removed in the PR comments as well: multiple labels on a form field no longer result in a failure as of the changes applied in https://github.com/dequelabs/axe-core/pull/1798

**Comment by jayaddison** [comment]
@brendankenny @patrickhulce From the test failures in this PR, upgrading `axe-core` to 3.4.0 pushes the lighthouse devtools bundle over the configured size limit of 400KB.

I'm not going to push any commits which change build limits, so was wondering if there's a good  way to resolve the failure in order to unblock the version bump?

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js]: hm, do we need to change our audit implementation to examine incompletes then? or just remove this audit entirely?

if I'm reading that commit correctly, it seems like this audit will never fail anymore. at which point, why have it

**Comment by jayaddison** [comment]
@patrickhulce Thanks for the quick response!

For the use case I'm looking at, it fixes an accessibility false-positive (https://github.com/dequelabs/axe-core/issues/1688).

The aria [spec](https://www.w3.org/TR/wai-aria-1.1/#combobox) considers a `combobox` having a child with `input[type=search]` as valid, whereas 3.3.0 marks a failure in that situation.  The [select2](https://github.com/select2/select2/tree/4.0.12) library renders DOM elements which fall into the failure case.

I hadn't noticed that the size increase was so significant; let me dig around and see if I can figure out what happened there.

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js (hunk 1 above)
Comment (jayaddison) [reply lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js]: That's my read of the situation too, yep.

**Comment by jayaddison** [comment]
@patrickhulce It looks like the majority of the bundle size increase is from locale data, much of which is new in 3.4.0.  From a slightly unscientific dir-by-dir diff and sort:

```
cat foo | sort
node_modules/axe-core/axe.d.ts 16K
node_modules/axe-core/axe.d.ts 6K
node_modules/axe-core/axe.js 644K
node_modules/axe-core/axe.js 784K
node_modules/axe-core/axe.min.js 344K
node_modules/axe-core/axe.min.js 404K
...
node_modules/axe-core/CHANGELOG.md 76K
node_modules/axe-core/CHANGELOG.md 88K
...
node_modules/axe-core/doc 584K
node_modules/axe-core/doc 608K
node_modules/axe-core/lib 6.3M
node_modules/axe-core/lib 6.4M
...
node_modules/axe-core/locales 148K
node_modules/axe-core/locales 244K
...
```
(identical file sizes omitted)

Let me see if the additional locale data would benefit `lighthouse`; if not, perhaps it can be excluded somehow.

**Comment by jayaddison** [comment]
@patrickhulce FWIW, another line of possibility is a change in one of `axe-core`'s dependencies:

https://github.com/dequelabs/axe-core/blob/v3.3.0/package.json#L89
vs
https://github.com/dequelabs/axe-core/blob/v3.4.0/package.json#L82
as per https://github.com/dequelabs/axe-core/pull/1707

This seems to result in a good chunk of extra code (basically an inlined import) in the output `axe.js`.

**Comment by jayaddison** [comment]
(didn't find anything conclusive on either of those, so some `git bisect` is happening)

**Comment by jayaddison** [comment]
Ok, my mistake earlier - in fact it is https://github.com/dequelabs/axe-core/pull/1829 that lead to the marked size increase in `axe-core` which then resulted in the overall bundle size failures here.

I'm still doing a little more investigation into the precise cause within that PR, and the value of the change to `lighthouse`, which may take a little more time given that I'm a bit new to each of them.

**Comment by jayaddison** [comment]
@patrickhulce Alright, yep - I've confirmed locally that it's the [introduction of core-js](https://github.com/dequelabs/axe-core/commit/683e005d3620f030b3fb24d3b309fb0d6583685d#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R90) in `axe-core` (which in turn was necessary for some [array functionality polyfilling](https://github.com/dequelabs/axe-core/commit/683e005d3620f030b3fb24d3b309fb0d6583685d#diff-91383c5b06b40827c141c40fd2a28787R18-R19) - cc @straker) that led to the majority of the size increase observed here.

Taking some time out here, but glad to help with any further investigation.

**Comment by patrickhulce** [comment]
FWIW I don't think 60KB is a *dealbreaker*, I just want to make sure we're getting something out of this tradeoff. The higher priority issue IMO is the now never-failing audit, really appreciate the investigation though we'd always like to help axe folks out with perf when we can :)


FYI @exterkamp apparently axe has a decent amount of locale data these days we might want to figure out how to leverage in our i18n'ing efforts.

**Comment by jayaddison** [comment]
@patrickhulce All makes sense, yep.  I think `incomplete` may be a relatively new concept (~late 2017), and implies neither a clear pass nor fail ([docs](https://github.com/dequelabs/axe-core/blame/72f0187dc8f534788fc0eb99a9ce48cd42a7923a/doc/API.md#L624)).

IIRC sometimes lighthouse displays grayed out advisory items which need to be checked manually?  If it's reasonably possible to transform the `multiple-labels` check into one of those, that might be an option (I'd probably open it as a PR branching off from here if so).

I didn't find any functional references to the `incomplete` return code in `lighthouse` right now, so I assume this would be a new code path if so.

**Comment by straker** [comment]
@jayaddison Incomplete can also mean the rule threw an error, in which case we report the error in the message of the incomplete result.

**Comment by jayaddison** [comment]
@straker Neat, thanks for the context.  Sounds like there's larger scope for comprehensively handling `incomplete` results, so as a new committer to `lighthouse`, I might politely bow out of designing & handling those (unless there are any small changes I can help with to improve things).

**Comment by connorjclark** [comment]
assuming that the inclusion of `core-js` introduced a lot of JS ... could we 1) build axe-core ourselves (and remove the polyfill stuff) or 2) convince axe to distribute a modern bundle in addition to their current one?

**Comment by straker** [comment]
@connorjclark won't take too much convincing https://github.com/dequelabs/axe-core/issues/1935 :)

**Comment by connorjclark** [comment]
Oh, great :)

**Comment by jayaddison** [comment]
Awesome :)

**Comment by patrickhulce** [comment]
alright this should be unblocked now with #10072 merging!

**Comment by jayaddison** [comment]
@patrickhulce Sometimes felt like I've been aiming for a record commit-to-line-diff ratio :)

Just one more `yarn type-check` issue to resolve which is showing up in `master` too I think:

```
lighthouse-core/audits/accessibility/axe-audit.js(51,7): error TS2322: Type 'Result' is not assignable to type 'Product'.
```

Looking into this currently.

**Comment by jayaddison** [comment]
I *believe* https://github.com/GoogleChrome/lighthouse/pull/10270 should resolve the type check failures.  I'm leaning on CI a little, having run some but not all tests locally.

**Pull request merged**